server/datastore/mysql/hosts.go

```diff
@@ -4426,15 +4426,20 @@ WHERE
 		counts = filtered
 	}
 
+	// filter by name and version
+	if name != nil && version != nil {
+		var filtered []fleet.OSVersion
+		for _, os := range counts {
+			if *name == os.NameOnly && *version == os.Version {
+				filtered = append(filtered, os)
+			}
```

Comment:
Why not combine this with the platform-filtering code above?

Reply:
good idea, that change is pushed